### ISSUE TITLE
Compile/Run Through Terminal/cmd

### ISSUE THREAD

**Issue by jessedevaney** [rgbkrk/atom-script#743]
I'm not familiar with your code but is there a way to change it so Atom doesn't capture the input and instead the file opens a new terminal/cmd prompt and compiles/runs inside that. I've been having issues trying to run code in Atom for programs that require user input and for some functions that return text not encoded using utf-8 which causes errors when atom tries to capture the terminal input. 

So is there a way to modify the code so it prompts terminal/cmd to open up and execute a command like 
*Open terminal/cmd* then execute 
1. "javac file.java"
2. "java file"
or for python
3. "python file.py"

**Comment by nixel2007** [comment]
I think we need a boolean parameter in build config (at grammars.coffee and at config window), like `shell: true`. So devs can specify where their code must be executed. 

Maybe we can get some useful code from the [build](https://github.com/noseglid/atom-build) package - it have this functionality. 

/cc @rgbkrk

**Comment by rgbkrk** [comment]
Yeah there's no problem with shelling out to open a brand new terminal. How about if the checkbox is:

* [ ] Open in new terminal

**Comment by jessedevaney** [comment]
Yeah, that would work. I'm guessing what I am asking is, is there anything I can do that would be a quick simple change that would allow me to edit the code my self? I'm not familiar with coffeescript.

**Comment by rgbkrk** [comment]
At the moment I don't think it's a quick simple change, especially cross platform.

I'm happy to help you start hacking on this package though in the coming week.

**Comment by jessedevaney** [comment]
thank you very much!

**Comment by jessedevaney** [comment]
When can I expect this to be done? Not to rush you or anything just would like an estimate if possible. :)

**Comment by rgbkrk** [comment]
Whoop! Do you have a good time today for a video chat and screenshare to pair on this?

**Comment by jessedevaney** [comment]
I haven't used coffeescript before. Maybe I made in unclear, I'm new to using github. I was wondering if it was a feature you could implement. I mean I would learn coffeescript, but I don't know how I would be of help to you? @rgbkrk or maybe I am misunderstanding you? sorry.

**Comment by rgbkrk** [comment]
Well, I was primarily saying that I'd love to help you learn how to work with this code base. This isn't a feature I want to devote any time to right now based on priorities and current workload. I'm always happy to donate a little bit of time in helping others learn though.

**Comment by Stormherz** [comment]
@rgbkrk I guess I could implement this (for example, we could place such optional checkbox in run options dialog and add global setting checkbox, which will allow running in external terminal on every script run), though there is a few questions pending - how can we determine terminal in a cross-platform way (specify in settings? guess by os?)

**Comment by rgbkrk** [comment]
That's definitely a can of worms. You can switch based on the OS, and at least got Windows check to see if powershell exists before launching cmd.exe

Not sure how you would know for some of the other operating systems honestly. Maybe this is a matter of specifying this as an arg to script options

**Comment by nixel2007** [comment]
There is some OS-checks In grammars.coffee, see "C" for example. Isn't this enough?

**Comment by rgbkrk** [comment]
Oh I mean the can of worms is in the launching of a terminal. I could be way wrong and I'd prefer that I am. :wink: If you're happy to code on this one, I'd love to see it!

**Comment by Stormherz** [comment]
Well, I'll start then and we'll see where it's coming =)

**Comment by sripathm2** [comment]
how can I input as a user while running a java program ?
For example if the code is -
import java.util.*;
public class h
{
public static void main(String[] args)
{
  Scanner ob=new Scanner(System.in);
  System.out.println("enter a number");
  int i=ob.nextInt();
System.out.println("hello"+i);
}
}

on applying shift+ctrl+b the program runs but I am not able to enter any input.

Please help.

**Comment by Ninja2k** [comment]
Textpad does it well, for Java it just calls the default command shell on Windows, this way every type of command is supported since they are built for use by the default shell,
